### PR TITLE
[platform] Add new platform as4630-54npe

### PR DESCRIPTION
**Pull request by seanwu-ec**
#### Why I did it
To support Accton's new model: AS4630-54NPE

#### How I did it
Part of this PR is restored from PR#6189. And the rest of the code is basically identical to AS4630-54PE's.

#### How to verify it
Built the image and tested it on DUT. The tests include some functional checks and going through some commands on the `show` program.

#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [ ] 202012
- [ ] 202106

#### Description for the changelog

[Accton] add new platform AS4630-54NPE

#### A picture of a cute animal (not mandatory but encouraged)
unnamed

### CODE REVIEW

**Comment by seanwu-ec** [comment]
> This pull request **introduces 1 alert** when merging [daebd20](https://github.com/Azure/sonic-buildimage/commit/daebd20f9233ed0e751514913eef917fa05a7e42) into [7698747](https://github.com/Azure/sonic-buildimage/commit/76987470280de3fde9cf96628c1abfec1283dc82) - [view on LGTM.com](https://lgtm.com/projects/g/Azure/sonic-buildimage/rev/pr-0a20a51f2975b3dc8106d22d2960b12553bddf65)
> 
> **new alerts:**
> 
> * 1 for Wrong name for an argument in a call

I don't know how come LGTM has this alert. namedtuple DOES have 'defaults' argument, and it works.
1. https://docs.python.org/3.7/library/collections.html#collections.namedtuple
2. https://github.com/python/cpython/blob/041bfafeb00b0e5e565986b01c00bf9cddce3b4c/Lib/collections/__init__.py#L316

**Comment by seanwu-ec** [comment]
Hi @lguohan , hi @jleveque , would you mind give me a hint about how to get my PRs reviewed and merged? Anything would do. Thanks.

**Comment by prgeor** [comment]
> This pull request **introduces 1 alert** when merging [dee8c37](https://github.com/Azure/sonic-buildimage/commit/dee8c3721283668fb512971be53f3ae1299b6a96) into [c80dabd](https://github.com/Azure/sonic-buildimage/commit/c80dabd0d825e09f4dc720dff2c3d3405aa64899) - [view on LGTM.com](https://lgtm.com/projects/g/Azure/sonic-buildimage/rev/pr-5188fa1f8ef6c21c528a291888dde683c32221f8)
> 
> **new alerts:**
> 
> * 1 for Wrong name for an argument in a call

@seanwu-ec new LGTM warnings

**Comment by seanwu-ec** [comment]
> > This pull request **introduces 1 alert** when merging [dee8c37](https://github.com/Azure/sonic-buildimage/commit/dee8c3721283668fb512971be53f3ae1299b6a96) into [c80dabd](https://github.com/Azure/sonic-buildimage/commit/c80dabd0d825e09f4dc720dff2c3d3405aa64899) - [view on LGTM.com](https://lgtm.com/projects/g/Azure/sonic-buildimage/rev/pr-5188fa1f8ef6c21c528a291888dde683c32221f8)
> > **new alerts:**
> > 
> > * 1 for Wrong name for an argument in a call
> 
> @seanwu-ec new LGTM warnings

Hi @prgeor,
Thanks for your review. 
As I commented above, I don't know how come LGTM has this alert. 
namedtuple DOES have 'defaults' argument, and it works. The same goes for Python3.9.

Plz check:
    https://docs.python.org/3.7/library/collections.html#collections.namedtuple
    https://github.com/python/cpython/blob/041bfafeb00b0e5e565986b01c00bf9cddce3b4c/Lib/collections/__init__.py#L316
Thanks.

**Comment by seanwu-ec** [comment]
Hi @prgeor, 
You just remind me that Accton has decided to keep that 6-levels thermal threshold APIs ONLY to Accton's customized SONiC instead of messing up community SONiC code. I've removed the use of namedtuple there and the corresponding LGTM alert is gone. Thanks.

**Review by prgeor — APPROVED** [review]

File: device/accton/x86_64-accton_as4630_54npe-r0/Accton-AS4630-54NPE/port_config.ini [hunk 1]
```diff
@@ -0,0 +1,55 @@
+# name          lanes                 alias             index   speed           autoneg
+Ethernet0       26                    thousandE1        1         1000           1
```
Comment (prgeor) [inline device/accton/x86_64-accton_as4630_54npe-r0/Accton-AS4630-54NPE/port_config.ini:2]: autoneg should be either 'on' or 'off'

**Comment by sujinmkang** [comment]
@seanwu-ec can you share the platform api test result and the outputs of show platform cli commands?

File: device/accton/x86_64-accton_as4630_54npe-r0/pmon_daemon_control.json [hunk 2]
```diff
@@ -0,0 +1,4 @@
+{
+    "skip_ledd": true,
+    "skip_pcied": true
```
Comment (sujinmkang) [inline device/accton/x86_64-accton_as4630_54npe-r0/pmon_daemon_control.json:3]: Why do you skip pcied for this platform?

**Review by prgeor — CHANGES_REQUESTED** [review]
@seanwu-ec could you address pending comments

File: device/accton/x86_64-accton_as4630_54npe-r0/pmon_daemon_control.json (hunk 2 above)
Comment (akenliu) [reply device/accton/x86_64-accton_as4630_54npe-r0/pmon_daemon_control.json:3]: There is no pcie devices, so we skip this daemon.

File: device/accton/x86_64-accton_as4630_54npe-r0/Accton-AS4630-54NPE/port_config.ini (hunk 1 above)
Comment (akenliu) [reply device/accton/x86_64-accton_as4630_54npe-r0/Accton-AS4630-54NPE/port_config.ini:2]: we will update this later.

Comment (akenliu) [reply device/accton/x86_64-accton_as4630_54npe-r0/Accton-AS4630-54NPE/port_config.ini]: Hi @prgeor 
chihhan123 helps to update the file port_config.ini. please review again.

**Comment by ec-michael-shih** [comment]
Dear Reviewer:


Testing result about show platform CLI command as follow:
Please feel free to let us know if there have some cmd result need to present.
I would appreciate that this new platform can merge into community master branch. 

Thank you very much.

```
show platform psustatus
root@sonic:~# show platform psustatus
PSU    Model       Serial              HW Rev      Voltage (V)    Current (A)    Power (W)  Status    LED
-----  ----------  ------------------  --------  -------------  -------------  -----------  --------  -----
PSU 1  YPEB1200AM  SC000X182126001247  N/A               54.25           0.00         0.00  OK        green
PSU 2  YPEB1200AM  SC000X182126001241  N/A               54.25           0.50        27.00  OK        green
```
```
root@sonic:~# show platform firmware status
Chassis              Module    Component    Version       Description
-------------------  --------  -----------  ------------  -------------------------
Accton-AS4630-54NPE  N/A       CPLD1        5             CPLD 1
                               BIOS         v47.01.02.00  Basic Input/Output System
```

```
root@sonic:~# show platform fan
  Drawer    LED          FAN    Speed    Direction    Presence    Status          Timestamp
--------  -----  -----------  -------  -----------  ----------  --------  -----------------
FanTray1  green       FAN-1F     100%      exhaust     Present        OK  20210713 18:25:53
FanTray1  green       FAN-1R     100%      exhaust     Present        OK  20210713 18:25:53
FanTray2  green       FAN-2F     100%      exhaust     Present        OK  20210713 18:25:53
FanTray2  green       FAN-2R     100%      exhaust     Present        OK  20210713 18:25:53
FanTray3  green       FAN-3F     100%      exhaust     Present        OK  20210713 18:25:53
FanTray3  green       FAN-3R     100%      exhaust     Present        OK  20210713 18:25:53
     N/A  green  PSU-1 FAN-1      47%      exhaust     Present        OK  20210713 18:25:54
     N/A  green  PSU-2 FAN-1      49%      exhaust     Present        OK  20210713 18:25:54
```

```
root@sonic:~# show platform summary
Platform: x86_64-accton_as4630_54npe-r0
HwSKU: Accton-AS4630-54NPE
ASIC: broadcom
ASIC Count: 1
Serial Number: 463054NPE2144010
Model Number: C0PZZ4654047A
Hardware Revision: N/A
```

```
root@sonic:~# show platform syseeprom
TlvInfo Header:
   Id String:    TlvInfo
   Version:      1
   Total Length: 174
TLV Name             Code      Len  Value
-------------------  ------  -----  -----------------------------
Product Name         0x21       19  AS4630-54NPE-O-AC-F
Part Number          0x22       13  C0PZZ4654047A
Serial Number        0x23       16  463054NPE2144010
Base MAC Address     0x24        6  98:19:2C:65:3B:D1
Manufacture Date     0x25       19  11/05/2021 00:04:01
Platform Name        0x28       29  x86_64-accton_as4630_54npe-r0
ONIE Version         0x29       13  2020.08.00.07
MAC Addresses        0x2A        2  256
Manufacturer         0x2B        6  Accton
Manufacture Country  0x2C        2  TW
Vendor Name          0x2D        8  Edgecore
Diag Version         0x2E       11  01.01.02.00
CRC-32               0xFE        4  0xF5347694

(checksum valid)
```

```
root@sonic:~# show platform temperature
             Sensor    Temperature    High TH    Low TH    Crit High TH    Crit Low TH    Warning          Timestamp
-------------------  -------------  ---------  --------  --------------  -------------  ---------  -----------------
PSU-1 temp sensor 1           28           80       N/A             N/A            N/A      False  20210713 18:28:54
PSU-2 temp sensor 1           26           80       N/A             N/A            N/A      False  20210713 18:28:54
      Temp sensor 1           27.5         64       N/A             N/A            N/A      False  20210713 18:28:54
      Temp sensor 2           28.5         80       N/A             N/A            N/A      False  20210713 18:28:54
      Temp sensor 3           25.5         80       N/A             N/A            N/A      False  20210713 18:28:54
```

**Comment by akenliu** [comment]
Hi @prgeor & @sujinmkang 
 We (edgecore members, ec-michael-shih & chihhan123) add more information for you.
Do you have any new advice about this PR#?
Could you help to review again or approve?

**Comment by sujinmkang** [comment]
@seanwu-ec can you please rebase this PR?

**Comment by akenliu** [comment]
> @seanwu-ec can you please rebase this PR?

sean resigned.
There is a message, Merging is blocked with requested changes, so I use his account to send a new "requested a review".
Do you have any advice regarding this PR#?
1. we use sean's account to do rebase?
2. we close this PR# and resend a new PR# that bases on latest master?
3. other ideas?

**Comment by seanwu-ec** [comment]
Close this due to there is a new one for the latest master code-base.
https://github.com/sonic-net/sonic-buildimage/pull/14293

**Pull request closed without merging**